pkg/model/firewall.go

```diff
@@ -521,3 +403,44 @@ func JoinSuffixes(src SecurityGroupInfo, dest SecurityGroupInfo) string {
 
 	return s + d
 }
+
+func (b *KopsModelContext) AddDirectionalGroupRule(c *fi.ModelBuilderContext, t *awstasks.SecurityGroupRule) {
+
+	name := generateName(t)
+	t.Name = fi.String(name)
+
+	klog.V(8).Infof("Adding rule %v", name)
+	c.AddTask(t)
+
+}
+
+func generateName(o *awstasks.SecurityGroupRule) string {
+
+	var target, dst, src, direction, proto string
+	if o.SourceGroup != nil {
+		target = fi.StringValue(o.SourceGroup.Name)
+	} else if o.CIDR != nil && fi.StringValue(o.CIDR) != "" {
+		target = fi.StringValue(o.CIDR)
+	} else {
+		target = "0.0.0.0/0"
+	}
+
+	if o.Protocol == nil || fi.StringValue(o.Protocol) == "" {
+		proto = "AllProtos"
+	} else {
+		proto = fi.StringValue(o.Protocol)
+	}
+
+	if o.Egress == nil || !fi.BoolValue(o.Egress) {
+		direction = "ingress"
+		src = target
+		dst = fi.StringValue(o.SecurityGroup.Name)
+	} else {
+		direction = "egress"
+		dst = target
+		src = fi.StringValue(o.SecurityGroup.Name)
+	}
+
+	return fmt.Sprintf("%v-%v-from-%v-to-%v-%v-%v", direction,
+		proto, src, dst, fi.Int64Value(o.FromPort), fi.Int64Value(o.ToPort))
```

Comment:
Any special reason for using "%v"?
```suggestion
	return fmt.Sprintf("%s-%s-%d-%d-from-%s-to-%s", direction,
		proto, fi.Int64Value(o.FromPort), fi.Int64Value(o.ToPort), src, dst)
```

Reply:
When we have `Ingress`, do we still need `To`, or is it intended that it refers to current SG?
Same for `Egress` and `From`.

Reply:
We need to mention both src and dst to avoid duplications. or are you referring to the word "from" itself?